### PR TITLE
feat(rr): allow update of admins in rr 

### LINKED ISSUE #1108: [BUG] Unable to update admins
## Problem description
reviewers are permanent at the time of review request creation and requestors are unable to update it. The backend for updating rr details is already done and includes admin so the work is purely frontend and involves
1. getting state from RHF
2. use a hook to ping our BE

## Definition of done
1. requestors can change admins **after** review is created and this change is propagated throughout (ie, newly added admins can approve; removed admins cannot)

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
See #1108 for an overview of the problem

Closes #1108 

## Solution
1. remove `stateManager` functionality and use `react-hook-forms` to manage state instead.
2. create new hooks + service api to ping the backend on update admin change.

**Breaking Changes**
- Requires the corresponding backend PR to be merged to work.

### CODE REVIEW

File: src/layouts/ReviewRequest/components/ManageReviewerModal/ManageReviewerModal.tsx [hunk 1]
```diff
@@ -11,34 +11,65 @@ import {
...
+    }
+  }, [isUpdateReviewRequestSuccess, successToast])
+
+  useEffect(() => {
+    if (isUpdateReviewRequestError) {
+      errorToast({
+        description: getAxiosErrorMessage(updateReviewRequestError),
+      })
+    }
+  }, [errorToast, isUpdateReviewRequestError, updateReviewRequestError])
```
Comment (alexanderleegs) [inline src/layouts/ReviewRequest/components/ManageReviewerModal/ManageReviewerModal.tsx:72]: Nit: should this be placed in the hook itself instead, via `onError`/`onSuccess`? It doesn't seem to require anything in particular from the component, and the error/success toast should be the same regardless of where it's called

**Review by alexanderleegs — COMMENTED** [review]
Small nit, but otherwise lgtm

File: src/layouts/ReviewRequest/components/ManageReviewerModal/ManageReviewerModal.tsx (hunk 1 above)
Comment (seaerchin) [reply src/layouts/ReviewRequest/components/ManageReviewerModal/ManageReviewerModal.tsx:72]: no real opinion here - i've just always been placing it inside the component cos i think that's where devs expect to see the toast + gives the component/page the ability to give a better error message in the event that it's a 500.

not too attached though; wdyt?

Comment (alexanderleegs) [reply src/layouts/ReviewRequest/components/ManageReviewerModal/ManageReviewerModal.tsx:72]: No particularly strong feelings either way also - I'm fine with leaving it as is!

Comment (seaerchin) [reply src/layouts/ReviewRequest/components/ManageReviewerModal/ManageReviewerModal.tsx:72]: will take path of least resistance

File: src/types/reviewRequest.ts [hunk 2]
```diff
@@ -32,10 +34,16 @@ export interface ReviewRequestInfo {
   reviewers: User[]
 }
 
-export interface ReviewRequestDto extends Omit<ReviewRequestInfo, "reviewers"> {
+export interface CreateReviewRequestDto
+  extends Omit<ReviewRequestInfo, "reviewers"> {
   reviewers: string[]
 }
 
+export type UpdateReviewRequestDto = SetOptional<
```
Comment (kishore03109) [inline src/types/reviewRequest.ts:42]: hmm quick clarification: why is `reviewers` field set to `optional`?

Comment (seaerchin) [reply src/types/reviewRequest.ts:42]: made `reviewers` a required property! thanks for pointing this out!

**Review by dcshzj — APPROVED** [review]
LGTM!

**Pull request merged**